src/client/replication_ddl_client.cpp

```diff
@@ -1738,5 +1738,27 @@ replication_ddl_client::set_max_replica_count(const std::string &app_name,
         configuration_set_max_replica_count_rpc(std::move(req), RPC_CM_SET_MAX_REPLICA_COUNT));
 }
 
+error_with<configuration_rename_app_response>
+replication_ddl_client::rename_app(const std::string &old_app_name, const std::string &new_app_name)
+{
+    if (old_app_name.empty() ||
+        !std::all_of(old_app_name.cbegin(),
+                     old_app_name.cend(),
+                     (bool (*)(int))replication_ddl_client::valid_app_char))
+        return FMT_ERR(ERR_INVALID_PARAMETERS, "invalid old_app_name.");
+    if (!std::all_of(new_app_name.cbegin(),
+                     new_app_name.cend(),
+                     (bool (*)(int))replication_ddl_client::valid_app_char)) {
+        return FMT_ERR(ERR_INVALID_PARAMETERS,
+                       "Not support format for new_app_name. You can use "
+                       "letters and numbers, and also char in ['.', ':', "
+                       "'_'].");
```

Comment:
How about simplified by "Only 0-9a-zA-Z.:_ are valid" ? You can append it to the message in the macro mentioned above.